### ISSUE TITLE
Validation Pipe doesn't work with latest class-validator and class-transformer

### ISSUE THREAD

**Issue by vizlibdk** [nestjs/nest#2235]
## I'm submitting a...
<pre><code>
[X] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior
Validation Pipe isn't validating properties correctly. If we pass a string e.g. `"123"` to property with decorator `@IsNumber()` validation returns `true`, and vice versa - value `123` in property with decorator `@IsString()` returns true. For `@IsBoolean()` and value `1` the same thing.

## Expected behavior
Validation Pipe should return `false` when we pass number value to the string property, and vice versa.

## Minimal reproduction of the problem with instructions
I've created simple controller and DTO:
```
import { Controller, Post, Body, UsePipes, ValidationPipe } from '@nestjs/common';
import { IsDefined, IsNumber, IsString } from 'class-validator';

export class TestDto {
    @IsDefined()
    @IsNumber()
    number: number;

    @IsDefined()
    @IsString()
    string: string;
}

@Controller()
export class TestController {
    @UsePipes(new ValidationPipe({ whitelist: true }))
    @Post('/test')
    test(@Body() body: TestDto): boolean {
        return true;
    }
}
```
then I've sent simple POST request with body:
```
{
	"number": "123",
	"string": 123
}
```
It returns `true` but should returns:
```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": [
        {
            "target": {
                "number": "123",
                "string": 123
            },
            "value": "123",
            "property": "number",
            "children": [],
            "constraints": {
                "isNumber": "number must be a number"
            }
        },
        {
            "target": {
                "number": "123",
                "string": 123
            },
            "value": 123,
            "property": "string",
            "children": [],
            "constraints": {
                "isString": "string must be a string"
            }
        }
    ]
}
```
**IMPORTANT!** I've changed version of `class-validator` and `class-transformer` from
```
"class-transformer": "^0.2.2",
"class-validator": "^0.9.1",
```
to previous minor (from my elder another NestJS project)
```
"class-transformer": "0.1.9",
"class-validator": "0.8.5",
```
and Validation Pipe works fine.

## What is the motivation / use case for changing the behavior?
Fix validation.

## Environment
<pre><code>
Nest version: 6.0.0

 
For Tooling issues:
- Node version: 11.7.0 
- Platform: Windows 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by lilacdev** [comment]
I just opened another issue while you were opening this one. #2237 . This is perhaps related, even if I use 6.3.0 .

**Comment by vizlibdk** [comment]
@kamilmysliwiec I've created repository: https://github.com/vizlibdk/nestjs-issue It has two branches: `01-issue` with `class-transformer` version `^0.2.2`, and `02-issue-fixed` with `class-transformer` version `0.1.9`.

**Comment by kamilmysliwiec** [comment]
It should be fixed with the latest class-transformer release :) (0.2.3)
